### PR TITLE
WT-9773 Fix mirrored table bulk load fail on cache pressure

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/format/bulk.c [hunk 1]
```diff
@@ -186,20 +190,24 @@ table_load(TABLE *base, TABLE *table)
...
+        /* Report on progress. */
+        if (report_progress)
+            track(track_buf, keyno);
+
         /*
          * When first starting up, restart the enclosing transaction every 10 operations so we never
          * end up with an empty object. After 5K records, restart the transaction every 5K records
          * so we don't overflow the cache.
+         *
+         * If we are loading a mirrored table, commit after every operation to ensure that we are
```
Comment (etienneptl) [inline test/format/bulk.c:203]: `commit after every operation` -> `commit periodically` sounds better IMO, after every operation sounds like we are committing every single one.

**Review by etienneptl — APPROVED** [review]

File: test/format/bulk.c [hunk 2]
```diff
@@ -186,20 +190,24 @@ table_load(TABLE *base, TABLE *table)
...
         }
 
+        report_progress = (keyno < 5000 && keyno % 10 == 0) || keyno % 5000 == 0;
+        /* Report on progress. */
+        if (report_progress)
+            track(track_buf, keyno);
+
         /*
          * When first starting up, restart the enclosing transaction every 10 operations so we never
          * end up with an empty object. After 5K records, restart the transaction every 5K records
```
Comment (sueloverso) [inline test/format/bulk.c:200]: This comment is now below and not clearly associated with the 5k usage in `report_progress`.

File: test/format/bulk.c [hunk 3]
```diff
@@ -186,20 +190,24 @@ table_load(TABLE *base, TABLE *table)
...
-            /* Report on progress. */
-            track(track_buf, keyno);
-
-            if (g.transaction_timestamps_config) {
-                bulk_commit_transaction(session);
-                committed_keyno = keyno;
-                bulk_begin_transaction(session);
-            }
+        if (g.transaction_timestamps_config && (report_progress || base != NULL)) {
+            bulk_commit_transaction(session);
```
Comment (sueloverso) [inline test/format/bulk.c:208]: Is there a noticeable performance hit or anything else with mirror tables and this change? I.e. could we keep the same "10 operations" and just avoid the 5k part?

**Review by sueloverso — APPROVED** [review]
I am approving but have a question and a request to move either the comment or the line of code the comment refers to.

File: test/format/bulk.c (hunk 3 above)
Comment (quchenhao) [reply test/format/bulk.c:208]: We can do that.

File: test/format/bulk.c (hunk 2 above)
Comment (quchenhao) [reply test/format/bulk.c]: [b39de35](https://github.com/wiredtiger/wiredtiger/pull/8496/commits/b39de35aeb45f988313485b9efc69c271ad13713)

File: test/format/bulk.c (hunk 3 above)
Comment (quchenhao) [reply test/format/bulk.c:209]: [b39de35](https://github.com/wiredtiger/wiredtiger/pull/8496/commits/b39de35aeb45f988313485b9efc69c271ad13713)

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

File: test/format/bulk.c (hunk 3 above)
Comment (quchenhao) [reply test/format/bulk.c:209]: I tried that and it is failing the test. I think we need to be conservative an revert to commit after each operation.

https://spruce.mongodb.com/task/wiredtiger_ubuntu2004_stress_tests_format_stress_test_3_patch_3fdcbdde0e500d84b0d4d49c6409274f0d78c74e_6376d2c31e2d1776f73aba56_22_11_18_00_33_46/logs?execution=0

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**